src/app/3d/qgsmaptoolclippingplanes.cpp

```diff
@@ -127,7 +127,34 @@ void QgsMapToolClippingPlanes::canvasReleaseEvent( QgsMapMouseEvent *e )
   if ( e->button() == Qt::LeftButton )
   {
     const QgsPointXY point = toMapCoordinates( e->pos() );
-    if ( mRubberBandPoints->numberOfVertices() == 2 )
+    // not a dynamic capture, we should return a line geometry
+    if ( mRubberBandPoints->numberOfVertices() == 1 && !mDynamicCapture )
+    {
+      QgsPointXY pt0 = *mRubberBandPoints->getPoint( 0, 0 );
+      QgsPointXY pt1 = point;
+
+      try
+      {
+        const QgsPointXY startPointMap3d = mCt->transform( pt0 );
+        const QgsPointXY endPointMap3d = mCt->transform( pt1 );
+        QgsVector vec( endPointMap3d - startPointMap3d );
+        vec = vec.normalized().perpVector();
+
+        const QVector<QgsPointXY> points3DMap( { startPointMap3d + vec, endPointMap3d + vec } );
+
+        // build a geometry and transform it back to 2d map canvas coordinates
+        QgsGeometry geom( new QgsLineString( points3DMap ) );
+        geom.transform( *mCt, Qgis::TransformDirection::Reverse );
+
+        emit finishedSuccessfully( geom, 0 );
```

Comment:
hmm... we are transforming from 3D map's CRS back to 2D map canvas CRS - but then in the other code in 3D, we are using that geometry unchanged?

I am thinking about the following approach:
- have a new small class that would encapsulate the whole cross section setup: i.e. the two endpoints and width - and it would be always in the 3D map CRS (at some point later we may want to add also Z range to define top and bottom limit). The class could have a function to e.g. generate QgsGeometry for the 2D canvas rubber band
- the map tool could call Qgs3DMapCanvas::setCrossSection( const QgsCrossSection & crossSection ), and the finishedSuccessfully() signal could be without and arguments, and it would only take care of raising the 3D widget back